modules/environment.nix

```diff
@@ -11,6 +11,25 @@ with lib;
 let
   cfg = config.environment;
   profileScript = pkgs.writeShellScript "profile" cfg.shell.profile;
+
+  requiredPackages = map (pkg: setPrio ((pkg.meta.priority or 5) + 3) pkg) [
+    pkgs.bashInteractive
+    pkgs.busybox
+  ];
+
+  makeShellWrapper = pkg: name:
+    let
+      script = pkgs.writeShellScriptBin name ''
+        set -n
+        source /etc/profile
+        exec ${pkg}/bin/${name} "$@"
+      '';
+    in "${script}/bin/${name}";
+
+  shells = if cfg.shell.enable then ({
+    bash = (makeShellWrapper pkgs.bashInteractive "bash");
```

Comment:
These should not be needed, I'll merge so that your stuff is fixed but these should be removed.